src/cosalib/cli.py

```diff
@@ -0,0 +1,119 @@
+# NOTE: PYTHONUNBUFFERED is set in cmdlib.sh for unbuffered output
+# pylint: disable=C0103
+
+import argparse
+import logging as log
+import os
+
+#: All valid logger levels
+LOG_LEVELS = ('warn', 'error', 'debug', 'info')
+
+
+class Cli:
+    """
+    Abstraction for executing commands from the cli.
+    """
+
+    def __init__(self, **kwargs):
+        """
+        Initializes the Cli instance.
+
+        :param kwargs: All keyword arguments which will pass to ArgumentParser
+        :type kwargs: dict
+        """
+        self.parser = argparse.ArgumentParser(**kwargs)
+        self.add_env_arg(
+            'COSA_LOG_LEVEL', 'info', '--log-level',
+            choices=LOG_LEVELS, help='Set the log level')
+
+    def add_env_arg(self, env_var, default, *args, **kwargs):
+        """
+        Adds an argument that can also pull it's value from an
+        environment variable.
+
+        :param env_var: The environment variable to look at
+        :type env_var: str
+        :env default: The default for the argument
+        :type default: any
+        :param args: Non keyword arguments to pass to add_argument
+        :type args: list
+        :param kwargs: Keyword arguments to pass to add_argument
+        :type kwargs: dict
+        """
+        if kwargs.get('help') is None:
+            kwargs['help'] = ''
+        kwargs['help'] = kwargs['help'] + ' (Env: {})'.format(env_var)
```

Comment:
Should there always be an environment variable override? It seems like that would be something optional in which case it might be better to do something like so:
```python
kwargs['help'] = '{}{}'.format(kwargs['help'], ' (Env: {})'.format(env_var) if env_var else '')
```

Reply:
When using `add_env_arg` there always will be an environment override available.

Reply:
Whoops, should have read the function name closer.

Using something like that you could probably merge the `add_env_arg` with `add_argument` and have `env_var` & `default` be named variables defaulted to `None`

Reply:
Done!